### PR TITLE
[youku] gracefully handle single failure when downloading playlist

### PR DESCRIPTION
**Pull request by junzh0u**
TL;DR: When downloading a playlist, a single failure shouldn't terminate the whole process.

Youku playlist can have broken video links. I believe this happens really rare, but I have one sample here: http://www.youku.com/playlist_show/id_21492543.html
This playlist contains a broken video link: http://v.youku.com/v_show/id_XNjUxNzQyNzEy.html

Since python set is unordered, when downloading this playlist, we can hit this video at a random point of the whole process, and crash.

This diff is just a rough idea. I'm open for opinions. I have several in my mind actually:
Should we have a flag guarding this feature?
Should we apply this feature to all extractors? If yes is there a better way to do that other than changing every extractor that support playlist?

[Review on Reviewable](https://reviewable.io/reviews/soimort/you-get/651)

### CODE REVIEW

**Comment by jackyzy823** [comment]
thanks @junzh0u .
New ideas are welcomed!
